typing/typecore.ml

```diff
@@ -1956,10 +1956,22 @@ let add_pattern_variables ?check ?check_as env pv =
          {val_type = pv_type; val_kind = Val_reg; Types.val_loc = pv_loc;
           val_attributes = pv_attributes;
           val_uid = Uid.mk ~current_unit:(Env.get_unit_name ());
+          val_bound_type_vars = generic_free_variables pv_type;
          } env
     )
     pv env
 
+let copy_pattern_variables ~old_env env pvs =
+  List.fold_right begin fun {pv_id} env ->
```

Comment:
I don't think that use of begin/end instead of parentheses in iterators is usual practice in the ocaml codebase

Reply:
Actually we use it quite a bit for the `with_local_level` wrappers.
I personally find it more readable, but this is a question of taste.